### PR TITLE
gateway

### PR DESCRIPTION
**Pull request by Nick5500**

### CODE REVIEW

File: gateway/nest-microservice/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+
+async function bootstrap() {
+  const PORT = 3007;
```
Comment (MaxLogvyniukFive) [inline gateway/nest-microservice/src/main.ts:5]: Use value from .env

File: gateway/nest-microservice/package.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (MaxLogvyniukFive) [inline gateway/nest-microservice/package.json:1]: Use yarn.lock. in project must be one lock file

File: gateway/nest-microservice/src/movies/dto/update-movie.dto.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { IsArray, IsInt, IsString } from 'class-validator';
+
+export class UpdateMovieDto {
```
Comment (MaxLogvyniukFive) [inline gateway/nest-microservice/src/movies/dto/update-movie.dto.ts:3]: DTO end of name

File: gateway/nest-microservice/src/movies/movie.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
+import { CreateMovieDto } from './dto/create-movie.dto';
+import { UpdateMovieDto } from './dto/update-movie.dto';
+
+@Injectable()
+export class MovieService {
+  private readonly serverUrl: string = 'http://localhost:3000/api/';
```
Comment (MaxLogvyniukFive) [inline gateway/nest-microservice/src/movies/movie.service.ts:9]: Use value from env

File: gateway/nest-microservice/src/movies/movie.service.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  constructor(private httpService: HttpService) {}
+
+  getAll() {
+    const axiosRes = this.httpService.get(this.serverUrl + 'movies');
```
Comment (MaxLogvyniukFive) [inline gateway/nest-microservice/src/movies/movie.service.ts:14]: Use template literal `{this.serverUrl}movies`

File: gateway/nest-microservice/src/movies/dto/update-movie.dto.ts (hunk 3 above)
Comment (Nick5500) [reply gateway/nest-microservice/src/movies/dto/update-movie.dto.ts:3]: ???

**Review by MaxLogvyniukFive — APPROVED** [review]

**Pull request merged**